### PR TITLE
Mahdiyeh/Fix: mlt/mf markets

### PR DESCRIPTION
**Pull request by mahdiyeh-fs**

### CODE REVIEW

File: src/common/util.js [hunk 1]
```diff
@@ -1,5 +1,19 @@
-function filterRestrictedSymbols(active_symbols) {
-  return active_symbols.filter(function(item) { return !/^(BOOM|CRASH|STP).+/i.test(item.symbol)});
+function isMalta(landing_company){
+  if (landing_company) {
+     if (landing_company.gaming_company) {
+        return landing_company.gaming_company.shortcode === 'malta';
+     } else {
+        return false
+     }
```
Comment (matin-binary) [inline src/common/util.js:7]: unnecessary `else` condition, you can simply `return false;`

File: src/instruments/instruments.es6 [hunk 2]
```diff
@@ -73,31 +102,31 @@ function onMenuItemClick(symbol, displayName) {
...
         .cached.send({ trading_times: new Date().toISOString().slice(0, 10) })
-        .then(function(data) {
+        .then(function (data) {
             chartable_markets = menu.extractChartableMarkets(data);
             refresh_active_symbols();
             liveapi.events.on('login', refresh_active_symbols);
             liveapi.events.on('logout', refresh_active_symbols);
-
+            
             return chartable_markets;
```
Comment (matin-binary) [inline src/instruments/instruments.es6:114]: Remove extra space 🙏🏼

File: src/instruments/instruments.es6 [hunk 3]
```diff
@@ -6,59 +6,88 @@ import { getSortedMarketSubmarkets } from '../common/marketUtils';
...
-                return m.submarkets.length !== 0;
-            });
+                local_storage.set('active_symbols', filtered_symbols);
 
-            markets = getSortedMarketSubmarkets(markets);
+                const active_markets = _(filtered_symbols).groupBy('market').map(function (symbols) {
+                            const filtered_symbols = symbols;
+                            const sym = _.head(filtered_symbols);
+                            const market = { name: sym.market, display_name: sym.market_display_name };
+                    market.submarkets = _(filtered_symbols).groupBy('submarket').map(function (symbols) {
```
Comment (mustofa-binary) [inline src/instruments/instruments.es6:34]: nit: pls help align

File: src/instruments/instruments.es6 [hunk 4]
```diff
@@ -6,59 +6,88 @@ import { getSortedMarketSubmarkets } from '../common/marketUtils';
 import "jquery-growl";
 import "common/util";
 
-function refresh_active_symbols() {
+const get_active_symbol = (landing_company, country) => {
+    const is_mf        = isMaltaInvest();
```
Comment (mustofa-binary) [inline src/instruments/instruments.es6:10]: seems duplicated, we can make function utils for these variables

File: src/trade/tradeMenu.es6 [hunk 5]
```diff
@@ -8,48 +8,86 @@ const show_error = (err) => {
...
 };
-const refresh_active_symbols = () => {
+
+const trade_messages = {
+   no_mf : () => "Binary options trading is not available in your financial account.".i18n(),
+};
+
+const get_active_symbol = (landing_company, country) => {
+   const is_mf        = isMaltaInvest();
+   const is_mlt       = landing_company ? isMalta(landing_company) : false;
```
Comment (mustofa-binary) [inline src/trade/tradeMenu.es6:18]: ^duplicated here

File: src/tradingtimes/tradingTimes.es6 [hunk 6]
```diff
@@ -149,17 +149,15 @@ const initTradingWin = ($html) => {
       const refresh = (data) => {
         const result = processData(menu.extractFilteredMarkets(data[0]));
         const active_symbols = local_storage.get('active_symbols');
-        const filtered_symbols = filterRestrictedSymbols(active_symbols)
```
Comment (mustofa-binary) [inline src/tradingtimes/tradingTimes.es6:152]: any reason why are we showing boom|crash|stp to webtrader?

**Review by mustofa-binary — CHANGES_REQUESTED** [review]

File: src/tradingtimes/tradingTimes.es6 (hunk 6 above)
Comment (mahdiyeh-fs) [reply src/tradingtimes/tradingTimes.es6:152]: we want to remove all symbol filtering from FE , as we did in binary.com and smartTrader

**Review by mustofa-binary — APPROVED** [review]
lgtm

**Review by mustofa-binary — APPROVED** [review]

**Review by matin-binary — APPROVED** [review]

**Pull request merged**